src/test_utils/hooks.ts

```diff
@@ -0,0 +1,68 @@
+import { DependencyList, EffectCallback, useEffect, useRef } from "react";
+
+/**
+ * `usePrevious` is a custom hook that returns the previous value of a given value
+ * @param value
+ * @param initialValue
+ * @returns - the previous value
+ */
+const usePrevious = <T>(value: T, initialValue: T) => {
+  const ref = useRef(initialValue);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
```

Comment:
Do we need to redeclare `usePrevious` if we have an existing one in the hooks folder? :0

Reply:
Great catch!

Reply:
Actually, nvm they are slightly different. One of them declares an initialValue Param.